### PR TITLE
BDT analysis

### PR DESCRIPTION
**Pull request by ValVau**
Added a BDT for further evaluation

### CODE REVIEW

File: workflow.py [hunk 1]
```diff
@@ -75,27 +77,12 @@ def count_objects(self, variation):
...
-        model_file = self.params.xgboost[self._year].model_file
-        
-        # print("XGBoost Model file: ", model_file)
-
-        # Create input data
-
-        # Evaluate the BDT score
-
-        bdt_score = np.zeros(len(self.events), dtype=np.float64)
+        model = lgb.Booster(model_file=self.params.LightGBM_model)
```
Comment (andreypz) [inline workflow.py:90]: Looks like the name of the model is taken from a config file in `params/xgboost.yaml`, right?
That is how I intended this. But the yaml file is not committed in you PR. Could you please add it as well.
You may also rename it to `lgb.yaml`, or just `ml_configs.yaml`.

**Comment by andreypz** [comment]
Could you move the `model.txt` and `model_DY.txt` into params directory. Maybe also rename them to something like `lgbm_model_Run2_ZLL_v1.txt`. 
 Also `LightGBM.py` better in `scripts` directory.

File: cfg_VHcc_ZLL.py [hunk 2]
```diff
@@ -196,8 +224,8 @@
         "met_pt": HistConf( [Axis(coll="MET", field="pt", bins=50, start=0, stop=200, label=r"MET $p_T$ [GeV]")] ),
         "met_phi": HistConf( [Axis(coll="MET", field="phi", bins=50, start=-math.pi, stop=math.pi, label=r"MET $phi$")] ),
 
-        "BDT": HistConf( [Axis(field="BDT", bins=20, start=-1, stop=1, label="BDT")],
-                         only_categories = ['SR_mumu_2J_cJ','SR_ee_2J_cJ','SR_ll_2J_cJ','SR_ll_2J_cJ_low','SR_ll_2J_cJ_high']),
```
Comment (andreypz) [inline cfg_VHcc_ZLL.py:207]: Is there a reason to remove the `only_categories` part?

**Comment by ValVau** [comment]
Please, review recent changes. There is a separate folder with models. The models are integrated and working.

**Comment by andreypz** [comment]
These will require additional packages to be installed.
Could you add a section in `Readme.md` with a conda/pip installs necessary for this tow work.

**Comment by andreypz** [comment]
I was able to install the needed packages (keras, tensorflow, etc.) Now the runner does not complain about missing packages. But it still crushes.
I think the version of the packages is important. So we need instructions on how to setup the environment from scratch: which packages to install and how (via conda or pip).

@ValVau could you please provide a list of packages with versions.

@uttiyasarkar could you please follow up on this PR and test that it works for you. Then merge, once it's working.

**Comment by ValVau** [comment]
Sure, I will provide you with the versions I have. In the meantime I can say, that I only used conda, no pip install. It is dangerous to mix them, so if you create the env using conda, you shall keep using conda in it. This is my experience

**Comment by uttiyasarkar** [comment]
Hi @ValVau could you please write a bit more detailed description of the PR so that we can keep track of what changes we made? Also it would be a bit more clear for me to understand the purpose of the PR

**Comment by ValVau** [comment]
> Hi @ValVau could you please write a bit more detailed description of the PR so that we can keep track of what changes we made? Also it would be a bit more clear for me to understand the purpose of the PR

Hi. The general purpose of this PR is new developments regarding BDT and DNN integration into VHcc analysis. The LightGBM.py file is used to perform training of a BDT or a DNN for a given set of signal and background files. Alternatively, separate BDTs or DNNs for the ZLL mode can be trained (low- and high-pt regions). Respectively, the models trained by me are added here. To integrate the existing models into the workflow, workflow and config files have been modified accordingly. As well as the signal sample added to the Run3.json to make further trainings possible for the other contributors. Should any further questions arise, feel free to ask me.

**Comment by andreypz** [comment]
I was able to setup software, from scratch, and then the code runs fine. I get DNN and BDT distributions filled.

Merging this PR, but a few updates are expected next from @ValVau 
- Create a `.yaml` file with parameters related to BDT and DNN files and use it instead of hardcoding the paths in the config file.
- Move the model files into a separate directory, not to clutter the repository.
UPD: Actually, I see this is done only for DNN models, but not for BDT. Should be also done. Need some structure, since we will have trainings for each year separately

**Comment by andreypz** [comment]
I suggest the following sturcture:
```
./MVA/
./MVA/training.py (former LightGBM)
./MVA/Readme.md -> a file that describes how to run the training
./MVA/BDT_models/
./MVA/DNN_models/
```

**Pull request merged**